src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -77,15 +77,13 @@ class UtxoState(override val persistentProver: PersistentBatchAVLProver[Digest32
 
     val txProcessing = ErgoState.execTransactions(transactions, currentStateContext)(checkBoxExistence)
     if (txProcessing.isValid) {
-      persistentProver.synchronized {
-        val mods = ErgoState.stateChanges(transactions).operations
-        val resultTry = Traverse[List].sequence(mods.map(persistentProver.performOneOperation).toList).map(_ => ())
-        ModifierValidator(stateContext.validationSettings)
-          .validateNoFailure(fbOperationFailed, resultTry)
-          .validateEquals(fbDigestIncorrect, expectedDigest, persistentProver.digest)
-          .result
-          .toTry
-      }
+      val mods = ErgoState.stateChanges(transactions).operations
+      val resultTry = Traverse[List].sequence(mods.map(persistentProver.performOneOperation).toList).map(_ => ())
```

Comment:
Why it is now allowed to perform operations in persistentProver concurrently?

Reply:
The only place `applyTransactions` is called from (aside of tests) is `applyModifier` function, where external lock on `persistentProver` already applied